service/src/test/org/apache/hive/service/cli/TestRetryingThriftCLIServiceClient.java

```diff
@@ -44,14 +44,12 @@
  * Test CLI service with a retrying client. All tests should pass. This is to validate that calls
  * are transferred successfully.
  */
-public class TestRetryingThriftCLIServiceClient {
+public class TestRetryingThriftCLIServiceClient extends AbstractThriftCLITest {
   protected static ThriftCLIService service;
-  private HiveConf hiveConf;
-  private HiveServer2 server;
 
-  @Before
-  public void init() {
-    hiveConf = new HiveConf();
+  @BeforeClass
+  public void init() throws Exception {
```

Comment:
@armitage420 , method should be static to when you annotate with @BeforeClass

Reply:
also please check the reported sonar issues 
- redundant throws, 
- can we do something better than thread sleep, maybe not in this PR

https://sonarcloud.io/project/issues?id=apache_hive&pullRequest=5205&issueStatuses=OPEN,CONFIRMED&sinceLeakPeriod=true

Reply:
@deniskuzZ I wanted to fix the commits along with this change.

Regarding the no valid location, it seems like it is happening post enabling tez as the default execution engine in [HIVE-27972](https://issues.apache.org/jira/browse/HIVE-27972).

Also, I can pick the redundant throws and thread sleep issues soon.

Reply:
have you tried setting

Reply:
@deniskuzZ Thanks so much for your help! I did try setting the jar directory. I also came across some more errors that needed to be fixed. initializing hiveconf with HiveConfForTest seemed like a cleaner approach. And now it is working in my local as well!

Created [HIVE-28602](https://issues.apache.org/jira/browse/HIVE-28602) for redundant exception throws and thread.sleep.

Thanks for the review!